File: Easy/53. Maximum Subarray/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int acc = nums[0];
        int max = nums[0];

        for (int i = 1; i < nums.length; i++) {
            acc = Math.max(acc + nums[i], nums[i]);
            max = Math.max(max, acc);
        }

        return max;
    }
}

File: Easy/53. Maximum Subarray/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
  let cur = -Infinity, max = -Infinity;

  for (let num of nums) {
    cur = Math.max(cur + num, num);
    max = Math.max(cur, max);
  }

  return max;
};
